packages/codegen-ui-react/lib/react-component-renderer.ts

```diff
@@ -53,13 +64,27 @@ export class ReactComponentRenderer<TPropIn> extends ComponentRendererBase<
   }
 
   protected renderOpeningElement(): JsxOpeningElement {
-    const propertyAttributes = Object.entries(this.component.properties).map(([key, value]) =>
-      buildOpeningElementProperties(value, key),
-    );
+    const localStateReferences = filterStateReferencesForComponent(this.component, this.stateReferences);
+    const propertyAttributes = Object.entries(this.component.properties).map(([key, value]) => {
+      if (key in localStateReferences) {
+        const stateName = getStateName({ componentName: this.component.name || '', property: key });
+        return buildOpeningElementProperties({ bindingProperties: { property: stateName } }, key);
+      }
+      return buildOpeningElementProperties(value, key);
+    });
     const eventAttributes = Object.entries(this.component.events || {}).map(([key, value]) =>
       buildOpeningElementEvents(value, key, this.component.name),
     );
-    const attributes = propertyAttributes.concat(eventAttributes);
+    const controlEventAttributes = Object.entries(localStateReferences)
+      .filter(([, { addControlEvent }]) => addControlEvent)
+      .map(([key]) =>
+        buildOpeningElementEvents(
+          { bindingEvent: getSetStateName({ componentName: this.component.name || '', property: key }) },
+          'change', // TODO: use component event mapping
```

Comment:
Is there a usecase where this isn't going to be `change`? I'm thinking TextField, CheckBoxField, etc. are all basically just listening to the value 'change' event. It doesn't make sense for this to be applied to anything that isn't a field, right?